tests/unit/amlb/datasets/file/test_file_dataloader.py

```diff
@@ -0,0 +1,345 @@
+import os
+
+import numpy as np
+import pandas as pd
+import pytest
+
+from amlb.resources import from_config
+from amlb.data import DatasetType
+from amlb.datasets.file import FileLoader
+from amlb.utils import Namespace as ns
+
+here = os.path.realpath(os.path.dirname(__file__))
+res = os.path.join(here, 'resources')
+
+
+@pytest.fixture(autouse=True)
+def file_config():
+    return from_config(
+        ns(
+            input_dir="my_input",
+            output_dir="my_output",
+            user_dir="my_user_dir",
+            root_dir="my_root_dir",
+        )
+    ).config
+
+
+@pytest.fixture()
+def file_loader(tmpdir):
+    return FileLoader(cache_dir=tmpdir)
+
+
+@pytest.mark.use_disk
+def test_load_binary_task_csv(file_loader):
+    ds_def = ns(
+        train=os.path.join(res, "kc2_train.csv"),
+        test=os.path.join(res, "kc2_test.csv"),
+        target="problems"
+    )
+    ds = file_loader.load(ds_def)
+    assert ds.type is DatasetType.binary
+    assert len(ds.features) == 22
+    assert len(ds.predictors) == 21
+
+    assert ds.target.name == ds_def.target
+    assert len(ds.target.values) == 2
+    assert ds.target.values == ["no", "yes"]
+    assert ds.target.data_type == 'string'   # file loader doesn't support categories yet when loading CSV files
+    assert ds.target.is_target
+    assert not ds.target.has_missing_values
+
+    assert all([p.data_type in ['int', 'float'] for p in ds.predictors])
+    assert all([p.values is None for p in ds.predictors])
+    assert not any([p.is_target for p in ds.predictors])
+    assert not any([p.has_missing_values for p in ds.predictors])
+
+    assert ds.train.path == ds_def.train
+    assert ds.test.path == ds_def.test
+    assert ds.train.data_path('csv') == ds.train.path
+    try:
+        ds.train.data_path('arff')       # file loader doesn't support file auto-conversion yet
+        pytest.fail("should have raised")
+    except ValueError as e:
+        assert "Dataset kc2_train is only available in csv format" in str(e)
```

Comment:
The [`pytest.raises`](https://docs.pytest.org/en/stable/reference.html?highlight=raises#pytest-raises) syntax was made for this :)

Reply:
good point, I never remember this one.